toolchain/check/generic.cpp

```diff
@@ -231,6 +241,41 @@ static auto MakeGenericEvalBlock(Context& context, SemIR::GenericId generic_id,
   return context.inst_block_stack().Pop();
 }
 
+// Builds and returns an eval block, given the list of canonical symbolic
+// constants that the instructions in the eval block should produce. This is
+// used when importing a generic.
+auto RebuildGenericEvalBlock(Context& context, SemIR::GenericId generic_id,
+                             SemIR::GenericInstIndex::Region region,
+                             llvm::ArrayRef<SemIR::InstId> const_ids)
+    -> SemIR::InstBlockId {
+  context.inst_block_stack().Push();
+
+  Map<SemIR::InstId, SemIR::InstId> constants_in_generic;
+
+  // For the definition region, populate constants from the declaration.
+  if (region == SemIR::GenericInstIndex::Region::Definition) {
+    auto decl_eval_block = context.inst_blocks().Get(
+        context.generics().Get(generic_id).decl_block_id);
+    for (auto inst_id : decl_eval_block) {
```

Comment:
```suggestion
    constants_in_generic.GrowForInsertCount(decl_eval_block.size());
    for (auto inst_id : decl_eval_block) {
```

Reply:
Done. I also added a small size for the map to further reduce allocations, with a TODO to work out a better / less arbitrary one, and I added another `GrowForInsertCount` before the second loop below, because that will also add entries to the map.